forte/data/readers/tests/html_reader_test.py

```diff
@@ -0,0 +1,175 @@
+            original_text = pack.get_original_text()
+            self.assertEqual(original_text, input)
+
+    @data(
+        (Span(0, 3), Span(7, 10), '<title>The Original Title </title>',
+         "strict"),
+        (Span(18, 22), Span(101, 105), '''<!DOCTYPE html PUBLIC "-//W34.01//EN">
+        <html<head>
+        <title//
+        <p ltr<span id=p>Text</span</p>
+        </>''', "relaxed"),
+        # # cases ending to or starting from between the span
+        (Span(15, 30), Span(60, 95), '''<!DOCTYPE html>
+        <h1 id="section1" class="bar">Section 1</h1>
+        <p class="foo">foo bar\nbaz blah </p>
+        <!-- cool beans! -->
+        <hr/>
+        <br>
+        <p><em>The <strong>End!</strong></em></p>
+        <p><em>error</p></em>weird < q <abc@example.com>''',
+         "forward"),
+        # before span starts
+        (Span(0, 3), Span(0, 3), 'Some text<title>The Original Title </title>',
+         "relaxed"),
+        (Span(0, 3), Span(0, 3), 'Some text<title>The Original Title </title>',
+         "strict"),
+        # after span ends # There's an issue with this #TODO (assign) mansi
+        # returns a span of (43, 35) which is wrong.
+        # (Span(28, 28), Span(43, 43),
+        #  'Some text<title>The Original Title </title>T',
+        #  "strict"),
+        # same begin and end
+        (Span(14, 14), Span(21, 21),
+         'Some text<title>The Original Title </title>',
+         "strict"),
+        (Span(14, 14), Span(21, 21),
+         'Some text<title>The Original Title </title>',
+         "relaxed"),
+        (Span(14, 14), Span(21, 21),
+         'Some text<title>The Original Title </title>',
+         "backward"),
+        (Span(14, 14), Span(21, 21),
+         'Some text<title>The Original Title </title>',
+         "forward")
+    )
+    def test_reader_original_span(self, value):
+        new_span, expected_orig_span, html_input, mode = value
+        for pack in self.pl1.process_dataset(html_input):
+            # Retrieve original text
+            original_text = pack.get_original_text()
+            self.assertEqual(original_text, html_input)
+
+            # Retrieve original span
+            original_span = pack.get_original_span(new_span, mode)
+            self.assertEqual(expected_orig_span, original_span)
+
+    @data(
+        ["<title>The Original Title </title>",
+         "<!DOCTYPE html><html><title>Page Title</title><body><p>This is a "
+         "paragraph</p></body></html>"],
+        ["<html>Test1</html>", "<html>Test12</html>", "<html>Test3</html>"]
+    )
+    def test_reader_caching(self, value):
+        count_orig = 0
+        for _ in self.pl1.process_dataset(value):
+            count_orig = count_orig + 1
+
+        num_files = len(os.listdir(self._cache_directory))
+
+        self.assertEqual(num_files, count_orig)
+
+        # Test Caching
+        count_cached = 0
+        for _ in self.pl2.process_dataset(value):
+            count_cached = count_cached + 1
```

Comment:
So the test only counts the number of files but not the content? I think we should compare the content to ensure caching is correct.

Reply:
Good Point! I'll add that